src/hotspot/share/opto/loopPredicate.cpp

```diff
@@ -662,6 +662,19 @@ bool IdealLoopTree::is_range_check_if(IfNode *iff, PhaseIdealLoop *phase, Invari
   if (offset && !invar.is_invariant(offset)) { // offset must be invariant
     return false;
   }
+#ifndef PRODUCT
+  if (offset && phase->has_ctrl(offset)) {
+    Node* offset_ctrl = phase->get_ctrl(offset);
+    // If the control of offset is loop predication promoted by previous pass,
+    // then it will lead to cyclic dependency.
+    // Previously promoted loop predication is in the same loop of predication
+    // point.
```

Comment:
Move these comments down to the path with the assert. I also suggest to add the following: "This situation can occur when pinning nodes too conservatively. - can we do better?"